cloudygames/views.py

```diff
@@ -5,111 +5,87 @@
+    permission_classes = (UserIsOwnerOrOperatorExceptUpdate,)
+
+    def get_queryset(self):
+        user = self.request.user
+        if(user.is_staff):
+            return GameOwnership.objects.all()
+        return GameOwnership.objects.filter(user=user)
+
 class GameSessionViewSet(viewsets.ModelViewSet):
     serializer_class = GameSessionSerializer
     filter_class = GameSessionFilter
+    permission_classes = (UserIsOwnerOrOperatorExceptUpdate,)
     
     def get_queryset(self):
         user = self.request.user
         if(user.is_staff):
             return GameSession.objects.all()
-        return GameSession.objects.filter(player=user)
+        return GameSession.objects.filter(user=user)
 
-    def put(self, request):
+    def create(self, request):
         serializer = GameSessionSerializer(data=request.data)
 
         if serializer.is_valid():
-            user = self.request.user
-            gameid = serializer.data['game']
+            game = serializer.validated_data['game']
+            user = serializer.validated_data['user']
 
-            try:
-                game = Game.objects.get(id=gameid)
-            except Game.DoesNotExist: # Case 1: Invalid game
-                return Response(status=status.HTTP_400_BAD_REQUEST)
-
-            try:
-                session = GameSession.objects.get(player=self.request.user, game=game) # Case 2: Already joined
-            except GameSession.DoesNotExist:
+            if game.id in GameOwnership.objects.filter(user=user).values_list('game__id', flat=True): # User owns the game
                 controller = GameSession.join_game(self, game)
-                if(controller == -1): # Case 3: Invalid Request
-                    return Response(status=status.HTTP_400_BAD_REQUEST)
-
-                #Create game session
-                session = GameSession.objects.create(game=game, player=user, controller=controller)
-                serializer = GameSessionSerializer(session)
-                return Response(serializer.data, status=status.HTTP_201_CREATED)
-
+                if(controller != -1):
+                    #Create game session
+                    session = GameSession.objects.create(game=game, user=user, controller=controller)
+                    serializer = GameSessionSerializer(session)
+                    return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(status=status.HTTP_400_BAD_REQUEST)
 
-    def delete(self, request, format=None):
-        data = json.loads(request.body.decode())
-
-        game = Game.objects.get(id=data['game'])
-        user = self.request.user
-        session = GameSession.objects.get(game=game, player=user)
+    def destroy(self, request, *args, **kwargs):
+        try:
+        	if(self.request.user.is_staff):
+        		session = GameSession.objects.get(id=kwargs['pk'])
+        	else:
+        		session = GameSession.objects.filter(user=self.request.user).get(id=kwargs['pk'])
```

Comment:
you can simplify this query to

    session = GameSession.objects.get(user=self.request.user, id=kwargs['pk'])